src/test/scala/nl.knaw.dans.easy.bagstore/component/BagProcessingSpec.scala

```diff
@@ -200,4 +201,53 @@ class BagProcessingSpec extends BagProcessingFixture {
       }
     }
   }
+
+  "getReferenceBags" should "fail when an empty refbag.txt is found in the bag" in {
+    // first create dir with empty refbag.txt
+    val bagDir = makeBagDirForRefbagTests("")
+
+    val bagId = BagId(UUID.randomUUID())
+    bagProcessing.getReferenceBags(bagDir.path, bagId) should matchPattern {
+      case Failure(e: InvalidBagException) if e == InvalidBagException(bagId, "the bag contains an empty refbags.txt") =>
+    }
+  }
+
+  it should "succeed if no refbag.txt is found in the bag" in {
+    val bagDir = File(testDir.toString) / "refbag"
+    bagDir.createDirectories()
+    bagDir.clear()
+    val bagId = BagId(UUID.randomUUID())
+    bagProcessing.getReferenceBags(bagDir.path, bagId) shouldBe a[Success[_]]
+  }
+
+  it should "fail when an refbag.txt with only whitespaces is found in the bag" in {
+    val bagDir = makeBagDirForRefbagTests("                           ")
+    val bagId = BagId(UUID.randomUUID())
+    bagProcessing.getReferenceBags(bagDir.path, bagId) should matchPattern {
+      case Failure(e: InvalidBagException) if e == InvalidBagException(bagId, "the bag contains an empty refbags.txt") =>
+    }
+  }
+
+  it should "succeed a non-empty refbag.txt is found in the bag" in {
+    val bagDir = makeBagDirForRefbagTests("refbag content")
+    val bagId = BagId(UUID.randomUUID())
+    bagProcessing.getReferenceBags(bagDir.path, bagId) shouldBe a[Success[_]]
+  }
+
+  private def makeBagDirForRefbagTests(content: String): File = {
+    val bagDir = File(testDir.toString) / "refbag"
+    bagDir.createDirectories()
+    bagDir.clear()
+
+    (bagDir / "refbags.txt")
+      .createFile()
+      .write(content)
+
+    (bagDir / "bagit.txt")
+      .createFile()
```

Comment:
Isn't the file created as soon as you write something? In other words isn't this line redundant? Same above.

Reply:
correct